sklearn/utils/_metadata_requests.py

```diff
@@ -938,7 +968,7 @@ def _get_param_names(self, *, method, return_alias, ignore_self_request):
                     )
         return res
 
-    def _route_params(self, *, params, method):
+    def _route_params(self, *, params, method, parent, caller):
```

Comment:
This is a bit weird that the different defintion of `_route_params` does not have the same order even if the class are different.

Would it makes sense to have the parameters order in the same order and if end if the additional parameters when there are some for a specific class.

Reply:
I agree, but this would be a separate PR for me. This PR is too complicated already and it's not intended to clean up everything that is weird here in one go.
Those two `_route_params` differed from each other before and it's not clear to me why.

Reply:
I now reordered the signatures to match each other. It was easier than I thought, since they were called with the param name and a limited number of times.